lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt

```diff
@@ -28,6 +29,40 @@ enum class TransactionType(val txPrefix: Felt) {
     L1_HANDLER(Felt.fromHex("0x6c315f68616e646c6572")), // encodeShortString('l1_handler')
 }
 
+/**
+ * The version of the transaction.
+ *
+ * The version is used to determine the transaction format and is used for transaction signing.
+ *
+ * Standard versions include [V0], [V1], [V2], and [V3]. These are utilized for regular transaction execution.
+ * Query versions [V1_QUERY], [V2_QUERY], and [V3_QUERY] should only be used for creating transactions to be used
+ * in queries that do not alter the chain state, as in methods like [Provider.simulateTransactions] and [Provider.getEstimateFee].
+ * Avoid using query versions for transactions intended for execution.
```

Comment:
> Avoid using query versions for transactions intended for execution.

I'm pretty sure this should not even be possible, and such tx would fail

Reply:
It's not, hence the warning 😅

Reply:
But the warning says "Avoid using query versions for transactions intended for execution.", which implies it is possible :P

Reply:
Which should convey to the user "it's possible from the SDK-side", but should not be done to .

Reply:
Which should convey to the user "it's possible from the SDK-side, but should not be done".
Should I rephrase it to:

```
Sending transaction with a query version for execution will result in a failure.
```

Reply:
Yes, explicit is better than implicit :+1: